### DISCUSSION TITLE
✗ Kustomization reconciliation failed: failed to decode Kubernetes YAML from /tmp/kustomization-4019278654/env/dev-values.yaml: missing Resource metadata <nil>

### DISCUSSION THREAD

**Discussion by parveen203** [fluxcd/flux2#4063] [Q&A] [answered]
@stefanprodan I am getting this error while reconciling to helm repo (https://github.com/techiescamp/helm-tutorial)

Following are the important files I am using and the repository structure

└── flux-system
      ├── gotk-components.yaml
      ├── gotk-sync.yaml
      ├── kustomization.yaml
└── Project A
     ├── gitrepo.yaml
     ├── kustomize.yaml
     ├── mysecret-nginx.yaml
     ├── nginx-cr.yaml
     ├── gitrepo.yaml
     ├── nginx-hr.yaml
     ├── nginx-crb.yaml
     ├── nginx-sa.yaml

##kustomize.yaml reference:
apiVersion: kustomize.toolkit.fluxcd.io/v1
kind: Kustomization
metadata:
  name: nginx-app
  namespace: nginx-ns
spec:
  interval: 2m0s
  path: ./
  serviceAccountName: nginx-sa
  prune: true
  sourceRef:
    kind: GitRepository
    name: nginx-app
    namespace: nginx-ns

##gitrepo.yaml reference:
apiVersion: source.toolkit.fluxcd.io/v1
kind: GitRepository
metadata:
  name: nginx-app
  namespace: nginx-ns
spec:
  interval: 1m0s
  ref:
    branch: main
  secretRef:
    name: mysecret-nginx
  url: https://github.com/techiescamp/helm-tutorial

Could you look into this and suggest something as I am new to fluxcd to create helm release

**Comment by kingdonb** [comment, marked as answer]
Hello @parveen203 

You didn't paste the error, so it's not clear what issue you experienced!

But it's generally not a good idea to use this path in Flux kustomizations, unless there is only one Flux kustomization for the whole repo. I think that you have more than one Flux kustomization pointed at this repo, so they need to point at different non-overlapping paths as Flux and Kustomize will recurse through the resources in the path, and they should not be managed in Flux multiple times:

```
path: ./
```

Please check out the docs, you can start with core concepts and work through getting started:

* https://fluxcd.io/flux/concepts/
* https://fluxcd.io/flux/get-started/

There are more advanced examples at the bottom of the get started guide. Please take some time to work through the examples in the docs, and welcome to ask more questions here, or in Slack if something is unclear!
